heron/statefulstorages/src/java/org/apache/heron/statefulstorage/localfs/LocalFileSystemStorage.java

```diff
@@ -134,6 +146,23 @@ public void dispose(String topologyName, String oldestCheckpointPreserved,
     }
   }
 
+  private void cleanCheckpoints(String path, int remaining) throws StatefulStorageException {
+    if (FileUtils.isDirectoryExists(path) && FileUtils.hasChildren(path)) {
+      File[] children = new File(path).listFiles();
+      Arrays.sort(children, Comparator.reverseOrder());
+
+      // only keep the latest N remaining files, delete others
+      for (int i = remaining; i < children.length; i++) {
```

Comment:
if we do:
for (int i = 0; i < children.length - remaining; ++i) {
}

we dont need Comparator, and the sorting might be slightly faster because the list might be likely to be in the same order already.

Reply:
done